From e9655dc456d65ee98ec6d806b4a2c45d0a02c0f8 Mon Sep 17 00:00:00 2001
From: Jeffy Chen <jeffy.chen@rock-chips.com>
Date: Thu, 19 Dec 2019 14:22:32 +0800
Subject: [PATCH 9/9] media: Enable mojo media when using v4l2 codec on linux

The newest chromium turns to use mojo video decoder instead of gpu video
decoder.

Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 src/3rdparty/chromium/media/media_options.gni | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/3rdparty/chromium/media/media_options.gni b/src/3rdparty/chromium/media/media_options.gni
index faf947880..ea4093575 100644
--- a/src/3rdparty/chromium/media/media_options.gni
+++ b/src/3rdparty/chromium/media/media_options.gni
@@ -149,7 +149,8 @@ declare_args() {
   # remote process (e.g. "utility" process, see |mojo_media_host|).
   enable_mojo_media =
       is_android || is_chromeos || is_mac || is_win || enable_library_cdms ||
-      (is_desktop_linux && use_vaapi) || (is_chromecast && !is_fuchsia)
+      (is_desktop_linux && use_vaapi) || (is_chromecast && !is_fuchsia) ||
+      (is_linux && use_v4l2_codec)
 
   # Enable the TestMojoMediaClient to be used in mojo MediaService. This is for
   # testing only and will override the default platform MojoMediaClient, if any.
@@ -208,7 +209,8 @@ if (enable_mojo_media) {
     ]
     _default_mojo_media_host = "gpu"
   } else if (is_chromeos || is_mac || is_win ||
-             (is_desktop_linux && use_vaapi)) {
+             (is_desktop_linux && use_vaapi) ||
+             (is_linux && use_v4l2_codec)) {
     _default_mojo_media_services = [ "video_decoder" ]
     _default_mojo_media_host = "gpu"
   }
-- 
2.20.1

